deltachat-ios/Chat/ChatViewController.swift

```diff
@@ -2390,6 +2392,31 @@ extension ChatViewController: BaseMessageCellDelegate {
         }
     }
 
+    private func selectProxy(url proxyURL: String) {
+        let host = dcContext.checkQR(qrCode: proxyURL).text1 ?? ""
+        let dcAccounts = DcAccounts.shared
```

Comment:
I'm not a 100% happy with using the singleton here, the alternative would've been to inject a `DcAccounts`-object and when I tried to, this kind of blew up in size and complexity.

Reply:
using `DcAccounts.shared` is totally fine, no need to pass that around, there really should be only one. if that changes, we need some refactoring anyways (but, really, having multiple DcAccounts objects  is not planned nor forseeable)